### PR TITLE
Attempt to migrate Heapster tests to prow

### PR DESCRIPTION
**Pull request by x13n**
/hold

### CODE REVIEW

**Comment by BenTheElder** [comment]
it would be create if we didn't need to make docker calls at runtime (IE after the ProwJob Pod has started), but if we do you'll want to mimic one of the other docker-in-docker jobs like `pull-kubernetes-cross` (we need to mount through the cache SSD to the docker graph and set an env to enable dind in the bootstrap image).

thanks for working on this! 😄

**Comment by krzyzacy** [comment]
@x13n any interests to make heapster can be built with bazel? xD

**Comment by BenTheElder** [comment]
@krzyzacy it can probably build without spinning up another container at runtime or using bazel, but some things need docker for other things like creating images. I'm not sure which this one needs.

**Comment by x13n** [comment]
@krzyzacy Not really, unless I have to;)

@BenTheElder Thanks for the tips!

**Comment by x13n** [comment]
Is there any way I can dry run this on prow?

**Comment by BenTheElder** [comment]
@x13n the best option is to leave the jenkins job as is and create a job with the same name but also append "-prow" or "-canary" and then set it to `always_run: false` and remove `all` from the trigger regex and you can manually `/test job-foo-canary` on a test PR.

**Comment by BenTheElder** [comment]
Also again, if this job is just using docker to do some building, we are *already* running in a docker image in CI so creating another docker container at runtime just for building in is highly discouraged. If you're building docker images during the job it would be helpful to know more about that so we can try to correctly cache only the base images.

**Comment by x13n** [comment]
As far as I can tell, heapster is not using docker to run tests, it uses it only when building a new heapster image. I changed this to stop running docker on docker.

> @x13n the best option is to leave the jenkins job as is and create a job with the same name but also append "-prow" or "-canary" and then set it to always_run: false and remove all from the trigger regex and you can manually /test job-foo-canary on a test PR.

@BenTheElder So, I have to merge this PR first in order to test it? I already created a separate job entry.

**Comment by BenTheElder** [comment]
I'm not sure if this will work as intended (not very familiar with the heapster tests) but it looks fine from our end.

**Comment by BenTheElder** [comment]
Let's give this a shake and fix anything else in a follow-up since it's manually triggered :-)
Thanks for working on this!

**Comment by x13n** [comment]
Thanks! I created https://github.com/kubernetes/heapster/pull/1922 for testing.

File: jobs/config.json [hunk 1]
```diff
@@ -10834,6 +10834,12 @@
       "sig-instrumentation"
     ]
   },
+  "pull-heapster-e2e-prow": {
+    "scenario": "execute",
```
Comment (krzyzacy) [inline jobs/config.json]: assume you'd need some args to execute here?

Comment (x13n) [reply jobs/config.json]: Isn't this just executing the k8s container specified in the other file?

Comment (krzyzacy) [reply jobs/config.json]: the previous job uses `"scenario": "kubernetes_heapster",` though

Comment (x13n) [reply jobs/config.json]: True, but kubernetes_heapster scenario runs a container via docker, while I wanted the prow version to run the same container in a k8s pod.

Comment (krzyzacy) [reply jobs/config.json]: ohhhh I see, the current image does not even checks out the heapster repo :-) I'll give it a patch

**Comment by x13n** [comment]
Thanks!

**Pull request merged**